cli-rs/src/main.rs

```diff
@@ -327,8 +327,8 @@ fn compile_rust<F: FieldElement>(
         None => powdr_riscv::Runtime::base(),
     };
 
-    if continuations && !runtime.has_submachine("poseidon_gl") {
-        runtime = runtime.with_poseidon();
+    if continuations {
+        runtime = runtime.with_poseidon_for_continuations();
```

Comment:
```suggestion
        // The runtime might already contain the Poseidon machine *without* continuations,
        // but this call overwrites it to the Poseidon machine *with* continuations.
        runtime = runtime.with_poseidon_for_continuations();
```

Reply:
so there's no actual overwriting, but it rather assumes the user hasn't tried to add poseidon